README.md

```diff
@@ -1,213 +1,158 @@
-  }
-};
 ```
-
-Now you can set the object to a JavaScript global, making it available in JavaScript code:
-
-```java
-// Attach our interface to a JavaScript object called Utf8.
-engine.set("Utf8", Utf8.class, utf8);
-
-String greeting = (String) engine.evaluate(""
-    // Here we have a hex encoded string.
-    + "var hexEnc = 'EC9588EB8595ED9598EC84B8EC9A9421';\n"
-    // Call out to Java to decode it!
-    + "var message = Utf8.fromHex(hexEnc);\n"
-    + "message;");
-
-Log.d("Greeting", greeting);
+@JsExport
+fun launchZipline() {
+  val zipline = Zipline.get()
+  zipline.bind<TriviaService>("triviaService", RealTriviaService())
+}
 ```
 
-### Calling JavaScript from Java
+Now we can start a development server to serve our JavaScript to any running applications that
+request it.
 
-You can attach a Java interface to a JavaScript global object, and call JavaScript functions
-directly from Java!  The same Java types are supported for function arguments and return
-values as the opposite case above.
+```
+./gradlew samples:trivia:trivia-js:jsBrowserProductionRun --info --continuous
+```
 
-#### Example
+Note that this Gradle won't ever reach 100%. That's expected; we want the development server to stay
+on. Also note that the `--continuous` flag will trigger a re-compile whenever the code changes.
 
-TODO fix this example, as it no longer works with QuickJS!
-You should still be able to understand what is going on, though.
+You can see the served application manifest at
+[localhost:8080/manifest.zipline.json](http://localhost:8080/manifest.zipline.json). It references
+all the code modules for the application.
 
-Imagine a world where we don't have [Okio's `ByteString`][okio]. Fortunately, there's a [Duktape
-builtin][dukdec] that allows us to convert hex-encoded strings back into UTF-8! We can easily set up a
-proxy that allows us to use it directly from our Java code. First, define a Java interface
-that declares the JavaScript methods you would like to call:
+In `jvmMain` we need write a program that downloads our Kotlin/JS code and calls it. We use
+`ZiplineLoader` which handles code downloading, caching, loading, and caching. We create a
+`Dispatcher` to run Kotlin/JS on. This must be a single-threaded dispatcher as each Zipline instance
+must be confined to a single thread.
 
-```java
-interface Utf8 {
-  String fromHex(String hex);
+```
+suspend fun launchZipline(dispatcher: CoroutineDispatcher): Zipline {
+  val zipline = Zipline.create(dispatcher)
+  val baseUrl = "http://localhost:8080/".toHttpUrl()
+  val loader = ZiplineLoader(
+    dispatcher = dispatcher,
+    httpClient = OkHttpZiplineHttpClient(baseUrl, OkHttpClient()),
+    embeddedDir = "/".toPath(),
+    embeddedFileSystem = FileSystem.RESOURCES,
+    cacheDbDriver = DriverFactory().createDriver(),
+    cacheDir = "zipline-cache".toPath(),
+    cacheFileSystem = FileSystem.SYSTEM,
+    cacheMaxSizeInBytes = 10 * 1024 * 1024,
+    nowMs = System::currentTimeMillis,
+  )
```

Comment:
I don't think this is the API anymore. Didn't the interceptor-y redesign land?

Reply:
This is still what the user-facing frontend of this API looks like.

I think this API is too big, with too many parameters and stuff that should be optional. I wanna work with @adrw to fix that. The examples here are real, though I think we could do better syntactically.